mopidy_mpd/protocol/album_art.py

```diff
@@ -0,0 +1,205 @@
+import logging
+from mopidy_mpd import protocol
+from urllib.request import urlopen
+from urllib.parse import urlparse
+from pathlib import PurePath
+
+logger = logging.getLogger(__name__)
+
+cover_cache = {}
+
+
+class _config:
+    config = {}
+    image_dir = ""
+
+    def load_config(context):
+        from mopidy_local import Extension as local_ext
```

Comment:
Mopidy-MPD does not depend on Mopidy-Local, nor should it. This won't end well on a system without Mopidy-Local installed.

Reply:
I noticed that too. In 1f30b8b I adjusted @gvc0's code so that `load_config()` and with it the import of `mopidy_local` would only be called if album art for a `/local/` track is queried, implying that `mopidy_local` is installed.

This however adds complexity, especially when dealing with unit testing (of `album_art.py`).

Reply:
On unit-testing: I'm currently integrating @gvc0's changes into `test_albumart.py`, as it let the patch-coverage drop to 72%, which is below the target percentage for codecov.

Reply:
> I adjusted @gvc0's code so that `load_config()` and with it the import of `mopidy_local` would only be called if album art for a `/local/` track is queried, implying that `mopidy_local` is installed.
> This however adds complexity, especially when dealing with unit testing (of `album_art.py`).
The reason for the dependency on Mopidy-Local is indeed to support local albumart, and also to make it not dependent on the HTTP service. The MPD albumart feature would be a disappointment if it doesn't support this local albumart I believe. Complexity could be reduced if we accept the dependency on HTTP.

Reply:
We need to avoid adding -Local as a dependency, that isn't going to fly. I'm open to making core API changes. We did that before with distinct and we can do it again here if it solves the problem properly.

It might be preferable to first merge binary responses and then solve support for Local afterwards. Although I appreciate Mopidy-Local might be your motivation. How well does this work with streaming services in general? It looks like it doesn't work at all, is that accurate?

Reply:
I agree going step by step might be the reasonable approach. The feature can be extended at any time through additional PRs :)

I've moved the `mopidy-local` support to a dedicated branch ([leso-kn:feature/album-art-mopidy-local](https://github.com/leso-kn/mopidy-mpd/tree/feature/album-art-mopidy-local)) and rewinded this PR to just binary responses and `albumart` support as described in the original PR description (any media that provides an artwork url) including the fixed Github Actions configuration.

Test coverage on `albumart.py` is now 100% with no flake warnings.

Reply:
> We need to avoid adding -Local as a dependency, that isn't going to fly. I'm open to making core API changes. We did that before with distinct and we can do it again here if it solves the problem properly.
> 
The -Local dependency is there because I needed to get the path of its image cache directory. This value is not in Mopidy config, it's a hard coded subdirectory in -Local. Moving that value to the config would take away some of the complexity and would be an improvement anyway I think. (Note this is why I asked about how to handle config values earlier on, but I continued due to lack of response).

Also the question on the HTTP dependency is still open. If you accept that, meaning local albumart will only work when HTTP is active, then again things become much simpler.

For me it's not clear yet what you mean by avoiding the -Local dependency. Does it mean the MPD code should not break if -Local is not installed, or do you mean I should interact with any of the -Local backend functions (and database). In the latter case HTTP access might be the way to go anyhow.

> It might be preferable to first merge binary responses and then solve support for Local afterwards. Although I appreciate Mopidy-Local might be your motivation. How well does this work with streaming services in general? It looks like it doesn't work with other backends at all?! Is that accurate?

I don't know how it works with other back-ends as I don't know how they reference albumart. But I made sure the code doesn't break with unrecognised calls, and HTTP URL call are handled correctly. Also here I think it's fair to take an agile approach and handle one backend at a time.

I can improve the code I made but I need some guidance.